source/sorted_table_model.ts

```diff
@@ -63,32 +65,105 @@ export class SortedTableModel extends TableModel {
    * @param columnOrder - The column sort order.
    */
   public constructor(source: TableModel, comparator?: Comparator,
-      columnOrder?: ColumnOrder[]) {
+    columnOrder?: ColumnOrder[]) {
     super();
+    this.translatedTable = new TranslatedTableModel(source);
+    if (comparator) {
+      this.comparator = comparator;
+    } else {
+        this.comparator = new Comparator();
+    }
+    if (columnOrder) {
+      this.order = columnOrder.slice();
+      this.sort(source);
+    } else {
+      this.order = [];
+    }
+  }
+
+  /** Marks the beginning of a transaction. In cases where a transaction is
+   *  already being processed, then the sub-transaction gets consolidated into
+   *  the parent transaction.
+   */
+  public beginTransaction(): void {
+    console.log('begin!');
+    if (this.transactionCount === 0) {
+      this.operations = [];
+    }
+    ++this.transactionCount;
+  }
+
+  /** Ends a transaction. */
+  public endTransaction(): void {
+    --this.transactionCount;
+    if (this.transactionCount === 0) {
+      this.dispatcher.dispatch(this.operations);
+    }
   }
 
   /** Returns the column sort order. */
   public get columnOrder(): ColumnOrder[] {
-    return null;
+    return this.order;
   }
 
   /** Sets the order that the columns are sorted by. */
-  public set columnOrder(columnOrder: ColumnOrder[]) {}
+  public set columnOrder(columnOrder: ColumnOrder[]) {
+    this.order = columnOrder;
+  }
 
   public get rowCount(): number {
-    return 0;
+    return this.translatedTable.rowCount;
   }
 
   public get columnCount(): number {
-    return 0;
+    return this.translatedTable.columnCount;
   }
 
   public get(row: number, column: number): any {
-    return null;
+    return this.translatedTable.get(row, column);
   }
 
   public connect(slot: (operations: Operation[]) => void):
-      Kola.Listener<Operation[]> {
-    return null;
+    Kola.Listener<Operation[]> {
+    return this.dispatcher.listen(slot);
```

Comment:
This is likely not going to work, we can revisit this in a future case. For now just return ```null```.